client/state/sites/selectors.js

```diff
@@ -10,3 +10,23 @@
 export function getSite( state, siteId ) {
 	return state.sites.items[ siteId ] || null;
 }
+
+/**
+ * Returns the slug for a site, or null if the site is unknown.
+ *
+ * @param  {Object} state  Global state tree
+ * @param  {Number} siteId Site ID
+ * @return {String}        Site slug
+ */
+export function getSiteSlug( state, siteId ) {
+	const site = getSite( state, siteId );
+	if ( ! site ) {
+		return null;
+	}
+
+	if ( site.options && site.options.is_redirect ) {
+		return site.options.unmapped_url.replace( /^https?:\/\//, '' );
+	}
+
+	return site.URL.replace( /^https?:\/\//, '' ).replace( /\//g, '::' );
+}
```

Comment:
@aduth without it some users will not be able to access the site that they want. `hasConflict` is set when it sees that a .com site has the same url as a jetpack site. It sets the primary domain of the .com site to the `something.wordpress.com` even the site is domain mapped for example. This allow the user to navigate calypso without conflicts. 

Since otherwise the user would not be able to land on the site that they want. Please keep it otherwise I am sure we will get someone telling us that they can't access the site.